### PR TITLE
Use "allowlist" and "denylist" instead of "whitelist" and "blacklist"

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] ~Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).~
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

With all that's going on in the world right now, it feels like an important time to do this and a poll on Slack showed support. And it matches our Code of Conduct on inclusivity.

Replacing "whitelist" with "allowlist" was easy. Replacing "blacklist" with "denylist" was slightly more nuanced as it depended if it was in an error message shown to users or something that required context - so I've gone for "disallow" as a verb in some places.

---

I can't run `brew tests` or `brew style` at the moment because after having played around with the Sorbet PR, my gems are messed up even after `brew update-reset` on `master`. I should figure out how to consistently fix that. This is _draft_ until CI is green.

### CODE REVIEW

File: Library/Homebrew/.rubocop.yml [hunk 1]
```diff
@@ -74,7 +74,7 @@ Naming/PredicateName:
   # can't rename these
   AllowedMethods: is_32_bit?, is_64_bit?
 
-# whitelist those that are standard
+# allowlist those that are standard
```
Comment (claui) [inline Library/Homebrew/.rubocop.yml:77]:
```suggestion
# allow those that are standard
```

File: Library/Homebrew/dev-cmd/create.rb [hunk 2]
```diff
@@ -101,12 +101,12 @@ def create
       fc.update_path
     end
 
-    # Don't allow blacklisted formula, or names that shadow aliases,
+    # Don't allow disallowed formula, or names that shadow aliases,
```
Comment (claui) [inline Library/Homebrew/dev-cmd/create.rb:104]:
```suggestion
    # Check for disallowed formula, or names that shadow aliases,
```

**Comment by claui** [comment]
@issyl0 Just noticed another couple of `white_list` occurrences in `Library/Homebrew/diagnostic.rb`.

**Comment by issyl0** [comment]
Thanks @claui, I think I've changed them all now.

**Pull request merged**